gusto/state.py

```diff
@@ -284,6 +284,10 @@ def setup_diagnostics(self):
             f = SteadyStateError(self, name)
             self.diagnostic_fields.append(f)
 
+        fields = set([f.name() for f in self.fields])
+        field_deps = [(d, set(d.required_fields).difference(fields),) for d in self.diagnostic_fields]
```

Comment:
We need to care about ordering here.  Because in parallel we need the schedule to come out the same, and the order of the field dependencies determines the order of the schedule.

So perhaps write:

```python
field_deps = [(d, sorted(set(d.required_fields).difference(fields))) for d in self.diagnostic_fields]
```
?